models/rule_helpers.go

```diff
@@ -91,7 +91,8 @@ type CreateRuleParams struct {
 // CreateRule persists alert Rule.
 func CreateRule(q *reform.Querier, params *CreateRuleParams) (*Rule, error) {
 	id := "/rule_id/" + uuid.New().String()
-	if err := checkUniqueRuleID(q, id); err != nil {
+	var err error
```

Comment:
why declare it here and not directly in if statement?
same in remove rule

Reply:
Because there are other similar statements in that function, and each will allocate a new error.  By predeclaring `err` variable we can just reuse it each time.

Reply:
In terms of you examples we choose between
```
if err := do1(); err != nil {
}
if err := do1(); err != nil {
}
```
and
```
var err error
if err = do1(); err != nil {
}
if err = do2(); err != nil {
}
```

Reply:
It's micro-optimization.
Anyway we are doing it at many places, like https://github.com/percona/pmm-managed/blob/PMM-2.0/models/agent_helpers.go#L331
Also in go srcs, e.g this test https://github.com/golang/go/blob/master/src/os/example_test.go#L32

Reply:
In general, I'm trying to minimize allocations when it doesn't reduce code clearness. I agree that this is not important here, so I can change, it's not that big a deal)

Reply:
The bigger issue with `if err :=` is shadowing – there are several variables with the same name. See also Go issue #377